### PR TITLE
Extract commom metrics stuff in an abstract metrics publisher

### PR DESCRIPTION
**Pull request by danielpassos**

### CODE REVIEW

File: core/src/main/java/org/aerogear/mobile/core/metrics/impl/DeviceMetrics.java [hunk 1]
```diff
@@ -18,7 +17,7 @@
     private final String platform;
     private final String platformVersion;
 
-    public DeviceMetrics(final Context context) {
+    public DeviceMetrics() {
```
Comment (aliok) [inline core/src/main/java/org/aerogear/mobile/core/metrics/impl/DeviceMetrics.java:20]: nice catch :)

File: core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java [hunk 2]
```diff
@@ -1,11 +1,76 @@
...
+    private final Metrics[] defaultMetrics;
+
+    public MetricsPublisher(final Context context) {
+        this.context = context;
+
+        defaultMetrics = new Metrics[] {new AppMetrics(context), new DeviceMetrics()};
+    }
+
+    /**
+     * Parse metrics into a JSONObject and add communs information for all metrics requests:
```
Comment (aliok) [inline core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java]: minor typo: communs --> common

File: core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java [hunk 3]
```diff
@@ -1,11 +1,76 @@
...
+     *
+     * - clientId - timestamp - type - app - device
+     *
+     * @param type the type of metrics
+     * @param metrics metrics
+     * @return a JSONObject
+     */
+    protected JSONObject parseMetrics(final String type, final Metrics[] metrics) {
```
Comment (aliok) [inline core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java]: let's rename it to createMetricsPayload or createMetricsJSONObject.
I am not a native speaker, but "parsing" is not what happens here.

**Review by aliok — COMMENTED** [review]
@danielpassos left very minor comments. let me know if I should merge this or if you would like to update anything.
thanks for the improvement!

**Comment by danielpassos** [comment]
Done!

**Pull request merged**